### DISCUSSION TITLE
TransducerToProbe transform and ImageTo transducerOriginPixel

### DISCUSSION THREAD

**Discussion by NesBarn** [PlusToolkit/PlusLib#1077] [Q&A]
Hello everyone,

I have some questions regarding the image to probe calibration process.

I am using NDI, an ultrasound probe from supersonic imagine and in an old xml (config file), I can find for a dedicated probe (Probe_L14-5_38):

-  This matrix and I do not know to what it corresponds. Is it the transducerToProbe matrix? Is it mandatory to put in the xml file? If yes, how may I get it?

<DisplayableObject Type="Model" ObjectCoordinateFrame="TransducerOrigin" Id="ProbeModel" File="Probe_L14-5_38.stl"
      ModelToObjectTransform="
        0	0	0	-118
        1	0	0	-4
        0	1	0	42
        0	0	0	1" />

- Also, there is the transform From="Image" To="TransducerOriginPixel"
      Matrix="
        1	0	0	-355
        0	1	0	86
        0	0	1	0
        0	0	0	1"
In my case, as I am using a frame grabber and a different ultrasound system, the image I have includes not only the US image but also information (type of probe...) so is the center of the frame of reference placed at the left corner of the overall image and then I have to know where the transducerOrigin Pixel is situated? Please see  the attachments to better understand what i mean

- Last, with the calibration I did (maybe false because of the matrix I didn't get with my own US probe), I don't know why but I have to multiply by 10 the matrix to have something relatively consistent, except the transformations..

<img width="433" alt="Capture" src="https://user-images.githubusercontent.com/97302106/228503146-f356cde1-02d2-44a4-a85f-ce6766210e89.PNG">

Thanks a lot!

**Comment by Sunderlandkyl** [comment]
You can find information on the coordinate systems here: http://perk-software.cs.queensu.ca/plus/doc/nightly/user/ApplicationfCalCoordinateSystemDefinitions.html.

**Reply by NesBarn** [reply]
Yes, I saw this one so for the transducer origin, it is ok but for my other questions, could you please give me a hint?
Thank you :)!

**Reply by Sunderlandkyl** [reply]
TransducerOriginPixel is indeed located at the center of the ultrasound image, at the location you have indicated.

ImageToProbe will take the coordinates of the image (pixels) and convert them to the coordinates of the tracker (mm). The scaling is required to convert between px/mm.

The Ultrasonix device is a bit special as the Ultrasonix device provides the ImageToTransducer matrix. that will transform pixels in the image to mm with the origin at the center of the probe, so that you don't need to recalibrate the probe when you change the depth. The TransducerToProbe (mm to mm) is the only matrix required in this case to go from the transducer to the EM tracking coordinates.

For other devices, you will need to calibrate the probe at a specific depth using ImageToProbe.

Does this answer your questions?

**Reply by NesBarn** [reply]
Thank you for your answers!
It is only the factor of 10 that remains unknown... 
I also have another question regarding my 3D reconstruction of a target: when I shift quickly the probe, I have in my reconstruction a shift and I was wondering if it is because of my ImageToProbe calibration that is not accurate. For information, during the calibration, I let the probe static over the phantom because when I tried to tilt the probe, I got a very high error of calibration... Does this problem speaks to you?
Thank you again for your great help!

**Reply by ungi** [reply]
If you get a higher error when moving the probe, then you probably have a temporal calibration (time synchronization) issue between the ultrasound stream and the tracker stream. Maybe check the section on temporal calibration on this page: http://perk-software.cs.queensu.ca/plus/doc/nightly/user/ApplicationfCal.html

It is still possible to have an error in your ImageToProbe calibration too. To estimate that error, you may use an alternative method to compute ImageToProbe calibration, e.g. using a tracked stylus-tip in the water, and compare the various calibrations. It is never possible to know the true ImageToProbe calibration, but the difference between repeated calibrations may give you a sense of precision.

**Reply by NesBarn** [reply]
Hi Ungi,
Regarding the temporal calibration, whenever I did it, no plots can be shown and I have this message" Reading temporal calibration configuration failed!| in E:\D\PSNPTeb\PlusApp\fCal\fCalMainWindow.cxx(330)".
Regarding the alternative you are talking about, it is like for example, fix the stylus tip at the corner of the ultrasound probe and then being able to compute the transformation matrix?
